python/operator/bot/operator_bot.py

```diff
@@ -131,6 +133,45 @@ def divulge_aliases(event):
                     )]
         return client.submit(commands)
 
+    @client.ledger_created(Chat.UserSettings)
+    def user_setting_change(event):
+        logging.info(f'On {Chat.UserSettings}')
+        archive_after = event.cdata['archiveMessagesAfter']
+        message = f"archive retention has been changed to `{archive_after}s`"
+        return client.submit(exercise_by_key(
+            Chat.Chat,
+            {'_1': client.party,
+             '_2': event.cdata['user']},
+            'Chat_PostMessage',
+            {
+                'poster': client.party,
+                'message': message,
+                'postedAt': f"{int(time.time())}"
+            }
+        ))
+
+    @client.ledger_created(Chat.ArchiveBotRequest)
+    def archive_bot_respond(event):
+        logging.info(f'On {Chat.ArchiveBotRequest}')
+        name = event.cdata['botName']
+        enabled = 'On' if event.cdata['enabled'] else 'Off'
+        message = f"Turning `{name}` {enabled}"
+        if event.cdata.get('message'):
+            message += f"\n{event.cdata['message']}"
+        commands = [exercise(event.cid, 'Archive', {}),
+                    exercise_by_key(
+                        Chat.Chat,
+                        {'_1': client.party,
+                         '_2': event.cdata['user']},
+                        'Chat_PostMessage',
+                        {
+                            'poster': client.party,
+                            'message': message,
+                            'postedAt': f"{int(time.time())}"
+                        }
```

Comment:
This block, meaning _send a message_, appears a couple times. Might not be a bad idea to factor it out into a function with a decent name.